cdap-common/src/main/java/co/cask/cdap/common/metadata/AbstractMetadataClient.java

```diff
@@ -538,6 +542,13 @@ private String constructPath(MetadataEntity metadataEntity) {
 
   private String addQueryParas(String path, MetadataEntity metadataEntity, @Nullable  MetadataScope scope,
                                boolean showCustom) {
+    // for all other queries except getMetadata called on EntityId we don't want aggregation of metadata for runIds.
+    // CDAP-13721
+    return addQueryParas(path, metadataEntity, scope, showCustom, false);
+  }
+
+  private String addQueryParas(String path, MetadataEntity metadataEntity, @Nullable  MetadataScope scope,
```

Comment:
seems like this (and the existing method) were meant to be called addQueryParams?

Reply:
Done.